examples/two_apps/src/lib.rs

```diff
@@ -52,13 +52,28 @@ where
         }
     }
 
-    fn update(&mut self, msg: Msg, _: &mut Env<CTX, Self>) -> ShouldRender {
+    fn update(&mut self, msg: Msg, env: &mut Env<CTX, Self>) -> ShouldRender {
         match msg {
             Msg::SendToOpposite(title) => {
                 self.activator.send_message(Msg::SetTitle(title));
             }
             Msg::SetTitle(title) => {
+                let context = env.context();
+                match title.as_ref() {
+                    "Ping" => {
+                        self.activator.send_message(Msg::SetTitle("Pong".into()));
+                    }
+                    "Pong" => {
+                        self.activator.send_message(Msg::SetTitle("Pong Done".into()));
+                    }
+                    "Pong Done" => {
+                        self.activator.send_message(Msg::SetTitle("Ping Done".into()));
+                    }
+                    _ => {
+                    }
+                }
                 self.title = title;
+                drop(context);
```

Comment:
isn't this called automatically?

Reply:
You are right, but I don't sure the optimizer won't return it immediately after initialization. But it's important to keep the lock till the end.